pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/InstanceDecommissionIntegrationTest.java

```diff
@@ -0,0 +1,162 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.integration.tests;
+
+import java.io.IOException;
+import org.apache.helix.AccessOption;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.linkedin.pinot.controller.helix.ControllerRequestURLBuilder;
+
+import static org.testng.Assert.*;
+
+/**
+ * Integration test for instance decommission
+ */
+public class InstanceDecommissionIntegrationTest extends OfflineClusterIntegrationTest {
+
+  @Test
+  public void testInstanceDecommission() throws Exception {
+    ControllerRequestURLBuilder urlBuilder = ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL);
+
+    // Fetch the list of the instances
+    JSONObject response = new JSONObject(sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forInstanceList()));
+    JSONArray instanceList = response.getJSONArray("instances");
+    assertEquals(instanceList.length(), 2, "Expected to have one broker and one server");
+
+    // Try to delete a server that does not exist
+    String deleteInstanceRequest = ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forInstanceDelete("aa");
+    try {
+      sendDeleteRequest(deleteInstanceRequest);
+      Assert.fail("Delete should have returned a failure status (404)");
+    } catch (IOException e) {
+      // Expected exception beacuse of 404 status code
+    }
+
+    // Get the server name
+    String serverName = "";
+    String brokerName = "";
+    for (int i = 0; i < instanceList.length(); i++) {
+      if (instanceList.get(i).toString().startsWith("Server_")) {
+        serverName = instanceList.get(i).toString();
+      } else if (instanceList.get(i).toString().startsWith("Broker_")) {
+        brokerName = instanceList.get(i).toString();
+      }
+    }
+
+    // Try to delete a live server
+    deleteInstanceRequest = ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forInstanceDelete(serverName);
+    try {
+      sendDeleteRequest(deleteInstanceRequest);
+      Assert.fail("Delete should have returned a failure status (409)");
+    } catch (IOException e) {
+      // Expected exception because of 409 status code
+    }
+
+    // Kill Server
+    stopServer();
+
+    // Try to delete a server whose information is still on the idealstate
+    try {
+      sendDeleteRequest(deleteInstanceRequest);
+      Assert.fail("Delete should have returned a failure status (409)");
+    } catch (IOException e) {
+      // Expected exception because of 409 status code
+    }
+
+    // Delete the table
+    sendDeleteRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forSegmentDeleteAllAPI("mytable", "offline")
```

Comment:
This deletes all segments, not the table itself. Either update the comment or the delete call.